File: Leetcode/String/3.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        answer = 0
        for i in range(len(s)):
            seen = []
            for j in range(i, len(s)):
                if s[j] in seen:
                    answer = max(answer, j - i)
                    break
                else:
                    seen.append(s[j])
                    answer = max(answer, len(seen))
        
        return answer

File: Leetcode/String/3.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function (s) {
  if (s.length <= 1) return s.length;

  let start = 0;
  let end = 1;
  let answer = 1;

  while (end < s.length) {
    let substring = s.slice(start, end + 1);
    let set = new Set(substring);
    if (end - start + 1 === set.size) {
      answer = Math.max(answer, end - start + 1);
      end++;
    } else {
      start++;
      end++;
    }
  }

  return answer;
};
